tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/integer_range_operators.bal

```diff
@@ -47,3 +47,232 @@ function testHalfOpenIntRangeAsArray(int startValue, int endValue) returns int[]
     }
     return returnArray;
 }
+
+function testClosedIntRangeOnIntSubTypes() {
+    int[] arr = [];
+    int arrIndex = 0;
+
+    int:Unsigned32 startValue1 = 12;
+    int:Unsigned32 endValue1 = 14;
+    object {
+        public function __iterator() returns object {
+                public function next () returns (record {| int value; |}?);
+        };
+    } rangeAsArray = startValue1 ... endValue1;
+    foreach var val in rangeAsArray {
```

Comment:
Additionally you can write this as `foreach int val in startValue1 ... endValue1 { ... }`. Can you convert few test cases to this format?